### ISSUE TITLE
Strings are being sent as Uint8Array rather than string; JSON is not parsed.

### ISSUE THREAD

**Issue by pearcetm** [feross/simple-peer#240]
From reading the documentation (and from experimenting with this library for a small project a couple of years previously) I was expecting that sending data of string type would arrive at the destination in the same format (or as a parsed object, if JSON).

However, it appears that all strings are being sent as Uint8Array currently. I have a very simple example right now:

```
var peer1 = new SimplePeer({initiator:false});
peer1.on('signal',function(data){
	peer2.signal(data);
});
var peer2 = new SimplePeer({initiator:true});
peer2.on('signal',function(data){
	peer1.signal(data);
});

peer1.on('connect',function(){ 
	console.log('peer1 connected!');
});
peer1.on('data',function(data){
	console.log(data);
});
peer2.on('connect',function(){
	console.log('peer2 connected!');
	setTimeout(function(){
		peer2.send('Testing');
	}, 1000);
});
```

The logged message is not a plain string, but rather a Uint8Array.

Is this a feature of the library, or implementation-dependent (i.e. browser dependent)? I'm using Chrome 64 on a Mac. Am I missing something?

**Comment by nazar-pc** [comment]
It should be `Buffer` rather than just `Uint8Array`.
You can try `data.toString()`, which should result in your original string.

**Comment by pearcetm** [comment]
It isn't `Buffer`, it actually is `Uint8Array` - so there is no `.toString()` method.

What I'm using is `String.fromCharCode.apply(null,data)` which does work... but I was just wondering why the connection was working that way instead of sending data as a string (or Buffer, or other expected type)

**Comment by RationalCoding** [comment]
Enable object mode in your peer configuration and strings will be preserved `{objectMode:true}’.

**Comment by nazar-pc** [comment]
If you run your example in Node.js, you'll see:
```
peer1 connected!
peer2 connected!
<Buffer 54 65 73 74 69 6e 67>
```

So it is actually a buffer and in browser it is also emulated. And `Buffer` actually extends `Uint8Array`, so its `toString()` is different from `Uint8Array`'s.

**Comment by RationalCoding** [comment]
You’ll need to parse JSON yourself from a string, simple-peer doesn’t do anything like that.

**Comment by pearcetm** [comment]
@RationalCoding thanks for pointing me to `objectMode`. I see that there is a description in the readme, which I had missed.

I was looking at the section for `send`:

> ### peer.send(data)
> Send text/binary data to the remote peer. data can be any of several types: String, Buffer (see buffer), TypedArrayView (Uint8Array, etc.), ArrayBuffer, or Blob (in browsers that support it).

If I may make a suggestion, I think it would be useful to mention `objectMode` in this section of the readme, otherwise people may have trouble understanding the default behavior. The current description in the options section is a bit buried and the wording doesn't make it as obvious as it could that sending strings is something that needs to be manually enabled.

As far as simple-peer doing JSON parsing automatically:

> ### peer.on('data', function (data) {})
> Received a message from the remote peer (via the data channel). JSON strings will be parsed and the resulting Object emitted.

I'm wondering if this also depends on `objectMode` being enabled, but I can't test it right now.

**Comment by RationalCoding** [comment]
You’re right, that should be documented.

I didn’t realize we documented automatically parsing JSON. That could actually be really frustrating if you were trying to send JSON as a string without wanting to parse it. I’ll see if we actually do...

**Comment by pearcetm** [comment]
@RationalCoding My testing (and a review of the source code) indicates that JSON parsing is not done on receipt of incoming data - even in objectMode - though it *is* done in the signaling context. I agree with you - it should be up to the app whether to parse JSON or not (it's not hard...).  So, I think the documentation just needs to be updated to reflect this.

Thanks in advance for clarifying the documentaion!

**Comment by RationalCoding** [comment]
The parsing mistake should be addressed in #249
